### ISSUE TITLE
confirming to FPNTextFieldDelegate leads to Build error 'duplicate symbols'  when build with other pods

### ISSUE THREAD

**Issue by mobiniusnagesh** [chronotruck/FlagPhoneNumber#104]
**Describe the bug**
I have couple of frameworks already included in my project, and now I'm adding 'FlagPhoneNumber' using cocoa pods.

After updating pods, app builds fine and I can use FlagPhoneNumber, but as soon as I confirm to FlagPhoneNumberDelegate to validate phone number; build fails

**Error Log**
`duplicate symbol '_kPhoneNumberMetaDataExpandedLength' in:
    /Users/nagesh/Library/Developer/Xcode/DerivedData/App-azmlnulobwzbyldoryfxtfkucgqc/Build/Intermediates.noindex/App.build/Debug-iphonesimulator/App.build/Objects-normal/x86_64/EDCMapboxViewController.o
    /Users/nagesh/Library/Developer/Xcode/DerivedData/App-azmlnulobwzbyldoryfxtfkucgqc/Build/Intermediates.noindex/App.build/Debug-iphonesimulator/App.build/Objects-normal/x86_64/EDCSortFilterViewController.o
duplicate symbol '_kPhoneNumberMetaData' in:
    /Users/nagesh/Library/Developer/Xcode/DerivedData/App-azmlnulobwzbyldoryfxtfkucgqc/Build/Intermediates.noindex/App.build/Debug-iphonesimulator/App.build/Objects-normal/x86_64/EDCMapboxViewController.o
    /Users/nagesh/Library/Developer/Xcode/DerivedData/App-azmlnulobwzbyldoryfxtfkucgqc/Build/Intermediates.noindex/App.build/Debug-iphonesimulator/App.build/Objects-normal/x86_64/EDCSortFilterViewController.o
duplicate symbol '_kPhoneNumberMetaDataCompressedLength' in:
    /Users/nagesh/Library/Developer/Xcode/DerivedData/App-azmlnulobwzbyldoryfxtfkucgqc/Build/Intermediates.noindex/App.build/Debug-iphonesimulator/App.build/Objects-normal/x86_64/EDCMapboxViewController.o
    /Users/nagesh/Library/Developer/Xcode/DerivedData/App-azmlnulobwzbyldoryfxtfkucgqc/Build/Intermediates.noindex/App.build/Debug-iphonesimulator/App.build/Objects-normal/x86_64/EDCAuditMapCollectionViewCell.o
duplicate symbol '_kPhoneNumberMetaDataExpandedLength' in:
    /Users/nagesh/Library/Developer/Xcode/DerivedData/App-azmlnulobwzbyldoryfxtfkucgqc/Build/Intermediates.noindex/App.build/Debug-iphonesimulator/App.build/Objects-normal/x86_64/EDCMapboxViewController.o
    /Users/nagesh/Library/Developer/Xcode/DerivedData/App-azmlnulobwzbyldoryfxtfkucgqc/Build/Intermediates.noindex/App.build/Debug-iphonesimulator/App.build/Objects-normal/x86_64/EDCAuditMapCollectionViewCell.o
duplicate symbol '_kPhoneNumberMetaData' in:
    /Users/nagesh/Library/Developer/Xcode/DerivedData/App-azmlnulobwzbyldoryfxtfkucgqc/Build/Intermediates.noindex/App.build/Debug-iphonesimulator/App.build/Objects-normal/x86_64/EDCMapboxViewController.o
    /Users/nagesh/Library/Developer/Xcode/DerivedData/App-azmlnulobwzbyldoryfxtfkucgqc/Build/Intermediates.noindex/App.build/Debug-iphonesimulator/App.build/Objects-normal/x86_64/EDCAuditMapCollectionViewCell.o
ld: 858 duplicate symbols for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)`
**Screenshots**
![Screenshot 2019-07-12 at 11 20 26 am](https://user-images.githubusercontent.com/17541624/61106013-600e1d00-a499-11e9-9594-4614214a6b7c.png)
![Screenshot 2019-07-12 at 11 19 49 am](https://user-images.githubusercontent.com/17541624/61106018-656b6780-a499-11e9-8d50-4a527c08eeb2.png)

**Smartphone (please complete the following information):**
 - Device: Simulator
 - OS: iOS 
 - Version 12
 
 **FlagPhoneNumber (please complete the following information):**
 - Version 0.7.6

**Additional context**
I have created a demo app only with 'FlagPhoneNumber' pods without any other pods, it works!

**Comment by grifas** [comment]
Did you try to clean your project + remove DerivedData repository and rebuild ?

**Comment by mobiniusnagesh** [comment]
@grifas I did, it didn't help. It seems issues started when 'FPNTextFieldDelegate' has been updated to support Objective-C. i.e adding "@objc" keyword for the delegate.
Because I downgraded to the version without "@objc" and it builds.

**Comment by grifas** [comment]
It seems your files (EDCMapboxViewController, EDCSortFilterViewController and EDCAuditMapCollectionViewCell) are duplicate so check in Build Phases and delete it.

**Comment by mobiniusnagesh** [comment]
`858 duplicate symbols for architecture x86_64 
clang: error: linker command failed with exit code 1 (use -v to see invocation)`

It's not only those files, it is saying all of my source files are duplicated.

**Comment by inottawa** [comment]
@mobiniusnagesh did you end up figuring out what was causing this. I have the same issue, and I believe it's because I use a mixed Objective-C Swift project. All of the `duplicate symbols` warnings are an files that I have a project-swift.h import in the objective-c class.

**Comment by mobiniusnagesh** [comment]
Yes.. I just downgraded the library to the version where it does not have ‘@objc’ annotation for the delegates.
I don’t remember the extract version, because I’m no longer working on that. 
Hope this helps

**Comment by wazzah-marham** [comment]
@mobiniusnagesh can you please share the version number you downgraded to?

**Comment by awais-shahid** [comment]
Any solution for duplicate symbols, i want to use in objective c project

**Comment by mdstone06** [comment]
I fixed the issue by forking the branch and removing the `@objc` marker from the delegate. I have a mixed ObjC/Swift project, but only need to use the library from Swift controllers. You can either check out my fork or make the same change yourself. I followed the instructions from [this Stack Overflow question](https://stackoverflow.com/questions/20936885/cocoapods-and-github-forks) to use my forked branch with CocoaPods
<img width="688" alt="Remove @objc from delegate" src="https://user-images.githubusercontent.com/23640886/158861185-59560a23-07cc-44b9-a7d4-82757875c520.png">
.
